client/dashboard/src/pages/mcp/MCPDetails.tsx

```diff
@@ -34,9 +33,19 @@ import {
   useRemoveOAuthServerMutation,
   useUpdateToolsetMutation,
 } from "@gram/client/react-query";
-import { Badge, Button, Grid, Stack } from "@speakeasy-api/moonshine";
+import {
```

Comment:
Reminders on some overall comments on UX here

- Keep the button disabled when the server is public and oauth is not applicable
- When public only show the external and proxy options
- When private only show the Gram option
- Remove the environment dropdown from the Gram OAuth model. It's not necesarry.